backend/api/models.py

```diff
@@ -1,56 +1,99 @@
+        ('4', '橡膠加工'),
+        ('5', '非金屬礦物（石材）'),
+        ('6', '食品'),
+        ('7', '皮革'),
+        ('8', '紡織'),
+        ('9', '其他')
+    ]
+    status_list = [
+        ('D','已舉報'),
+        ('F','資料不齊'),
+        ('A','待審核')
+    ]
+
+    # All  Features
+    id =  models.UUIDField(max_length=36, primary_key=True, serialize=False, verbose_name='ID')
+    fact_name = models.CharField(max_length=50, blank=True, null=True)
+    lat = models.FloatField(max_length=16)
+    lng = models.FloatField(max_length=16)
+    landcode = models.CharField (max_length=50, blank=True, null=True)
+    fact_type = models.CharField(max_length=3, choices=fact_type_list)
+    status = models.CharField(max_length=1, choices=status_list)
+    status_time = models.DateField()
+    others = models.CharField(max_length=1024, blank=True)
+    creat_at = models.DateTimeField(auto_now_add=True)
+
+    # % Discussion!!
+    # fact_no = models.AutoField(max_length=36, unique=True, serialize=True, verbose_name='Factory No.')     Readable id (unique)
+    # update_time =  models.DateField(auto_now=True)   The last time of the change
 
-    This table should store information of a factory,
-    containing:
-    - name
-    - type: Enum, ref: https://g0v.hackmd.io/1w_44QhqTWKi2dcyzCitkA#%E5%B7%A5%E5%BB%A0%E5%88%86%E9%A1%9E
-    - position(lat, long),
-    - land index
-    - status
-    - reported_at (aka created_at)
-    - etc...
-
-    """
-    # TODO: write a migration for data initialization, ref: https://docs.djangoproject.com/en/2.2/howto/initial-data/
+    # TODO: write a migration for data initialization, ref: https://docs.djangoproject.com/en/2.2/howto/initial-data/ 
+      # % I still have no idea about this step. I have to make a example by myself without using the command "makemigration" because the DB is not ready?
+    
     point = models.PointField(srid=settings.POSTGIS_SRID)
 
 
 class ReportRecord(models.Model):
     """Report records send by users.
-
-    This table should contain the report record after calling these API:
-    - PUT /factories/{id}/{attribute}
-    - POST /factories
-    Necessary fields:
-    - foreign key to Factory
-    - foreign key to User (point to a null user at first)
-    - created_at
-    - etc...
-
-    `ReportRecord` will be queried in advanced by admins from
-    Citizen of the Earth, Taiwan. They will filter the most recent
-    records out every a few weeks to catch the bad guys.
-    """
+        `ReportRecord` will be queried in advanced by admins from
+        Citizen of the Earth, Taiwan. They will filter the most recent
+        records out every a few weeks to catch the bad guys.
+        Containing:
+        - id : Foreign Key related to "Factory"
+        - user_ip
+        - action_type :  (e.g. PUT)
+        - action_body :  The request
+        - created_at
+        - contect :  Could be emall, phone number, nick name and so on
+       """
+    id = models.ForeignKey('Factory', on_delete=models.PROTECT)
```

Comment:
```suggestion
    factory = models.ForeignKey('Factory', on_delete=models.PROTECT)
```

this is the convention. With it, you may reference it in a very pythonic way:
```python
report_record = ReportRecord(...)
factory = report_record.factory  # factory will be a Factory object
```

Reply:
Also, add a default `id` field:

```python
id = models.AutoField(primary_key=True)
```